test/ExecutionFramework.h

```diff
@@ -275,7 +277,13 @@ class ExecutionFramework
 	solidity::frontend::RevertStrings m_revertStrings = solidity::frontend::RevertStrings::Default;
 	solidity::frontend::OptimiserSettings m_optimiserSettings = solidity::frontend::OptimiserSettings::minimal();
 	bool m_showMessages = false;
+
+	std::shared_ptr<EVMHost> m_evmcHost;
```

Comment:
Why are there three?

Reply:
`m_evmcHost` is the "current" evmc host that is used for all interactions. The two others are an `evm1` evmc host `m_evmHost` and the `ewasm` evmc host `m_ewasmHost`. The logic is basically to set `m_evmcHost` to `m_evmHost` or `m_ewasmHost`, depending on what kind of evmc vm should be used and what vm's where defined. 

The switch is mainly done in `ExecutionFramework::reset(bool useEwasm)`, if `useEwasm` is true and there was a evmc vm defined supporting ewasm, `m_evmcHost` will be set to `m_ewasmHost`. If `useEwasm` is false, `m_evmcHost` is set to `m_evmHost`.

Reply:
Ah, ok, I see. Can you make it a plain (non-owning) pointer then?